baselines/ppo1/pposgd_simple.py

```diff
@@ -133,29 +143,34 @@ def learn(env, policy_fn, *,
         for (oldv, newv) in zipsame(oldpi.get_variables(), pi.get_variables())])
     compute_losses = U.function([ob, ac, atarg, ret, lrmult], losses)
 
-    U.initialize()
+    initialize_uninitialized(tf.get_default_session())
+    #U.initialize()
     adam.sync()
 
     # Load existing model
     if load_model:
         U.load_state(os.path.join(model_dir, "model"))
 
+    # Reinitialize exploration
+    #reinitialize_exploration(tf.get_default_session())
```

Comment:
This is a technique I called exploration restarter. In my experiments, I found that gaussian exploration is the most important hyperparameter, even more than learning rate. However, in PPO it tends to shrink fast and converge prematurely. Thus, I create this exploration restarter which restart exploration rate using the initial value defined in the network, instead of simply load the learned values from previous training. It worked well, mainly when the learning rate and exploration rate do not have the same magnitude. 

As you can see, I let this commented because it is not the default behavior, but it can be improved with a simple flag.

Reply:
Nice idea!